examples/pi4_mesa_build/README.md

```diff
@@ -0,0 +1,103 @@
+
+## Running the Pi headless
+
+By far the easiest way to get up and running with a Raspberry Pi is to configure it for headless operation. This removes the requirement to have a monitor, keyboard or mouse. To run headless the Pi needs access to the internet and for SSH enabled. The following guides from the Raspberry Pi foundation should help.
+
+- [Setting up a Raspberry Pi headless](https://www.raspberrypi.org/documentation/configuration/wireless/headless.md)
+- [SSH (Secure Shell)](https://www.raspberrypi.org/documentation/remote-access/ssh/)
+
+## Ensure all packages are using the latest version
+
+```
+sudo apt-get update
+sudo apt-get upgrade
+```
+
+## Install dependencies for building mesa and running Vulkan Kompute
+
+```
+sudo apt-get install \
+    git build-essential cmake \
+    python3-dev python3-mako python3-venv \
+    flex bison meson ninja-build \
+    libxcb-shm0-dev libxcb1-dev libxcb-*-dev \
+    libx11-dev libx11-xcb-dev x11proto-dri2-dev x11proto-dri3-dev \
+    libdrm-dev libxshmfence-dev libxrandr-dev libxfixes-dev \
+    vulkan-tools libvulkan-dev
+```
+
+## Clone mesa repository
+
+```
+git clone --depth 1 https://gitlab.freedesktop.org/mesa/mesa.git
+```
+
+## Build mesa
+
+Use meson and ninja to build mesa using the Broadcom Vulkan drivers. For information on the Gallium drivers please see [V3D — The Mesa 3D Graphics Library latest documentation](https://docs.mesa3d.org/drivers/v3d.html).
+
+TODO: Experiment adding kmsro and vc4 gallium drivers
+
+```
+meson --libdir lib \
+    --prefix /mesa-install \
+    -D platforms=x11 \
+    -D vulkan-drivers=broadcom \
+    -D gallium-drivers=v3d \
+    -D dri-drivers=[] \
+    -D buildtype=debug \
+    build
+
+ninja -C build
+sudo ninja -C build install
+```
+
+## Configure preferred Vulkan driver
+
+Export the path for the Broadcom drivers. This command will need to be run for every new terminal session. Alternatively adding this line to `~/.bash_profile` (os similar) will export the environment variable for every terminal session.
+
+```
+export VK_ICD_FILENAMES=/mesa-install/share/vulkan/icd.d/broadcom_icd.armv7l.json
+```
+
+To confirm that mesa was configured and built correctly use.
+
+```
+vulkaninfo
+```
+
+## Clone Vulkan Kompute
+
+Clone Vulkan Kompute for access to the latest Python tests.
+
+```
+git clone https://github.com/EthicalML/vulkan-kompute.git
+```
+
+## Install dependencies to run the tests 
+
+Navigate to the available tests and install required dependencies.
+
+```
+cd vulkan-kompute/python/test
+python3 -m venv .venv
+source .venv/bin/activate
+pip install --upgrade pip wheel
+pip install -r requirements-dev.txt
+pip install git+git://github.com/EthicalML/vulkan-kompute.git@master
```

Comment:
You can also install by running `pip install . -v` form the top  level directory

Reply:
May also be worth installing with a tag (for now perhaps master is good enough until i do te 0.7.0 release - can do it this week)

Reply:
When attempting to install from a relative path (either in the root or tests folder) I get the following error:

```
    running install
    running build
    running build_ext
    -- The C compiler identification is GNU 8.3.0
    -- The CXX compiler identification is GNU 8.3.0
    -- Check for working C compiler: /usr/bin/cc
    -- Check for working C compiler: /usr/bin/cc -- works
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- Check for working CXX compiler: /usr/bin/c++
    -- Check for working CXX compiler: /usr/bin/c++ -- works
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    -- Found Vulkan: /usr/lib/arm-linux-gnueabihf/libvulkan.so
    CMake Error at src/CMakeLists.txt:66 (add_subdirectory):
      The source directory

        /tmp/pip-req-build-flcbpbf5/external/Vulkan-Headers

      does not contain a CMakeLists.txt file.


    CMake Error at src/CMakeLists.txt:67 (get_target_property):
      get_target_property() called with non-existent target "Vulkan-Headers".


    CMake Error at src/CMakeLists.txt:78 (add_subdirectory):
      The source directory

        /tmp/pip-req-build-flcbpbf5/external/fmt

      does not contain a CMakeLists.txt file.


    CMake Error at src/CMakeLists.txt:142 (add_subdirectory):
      The source directory

        /tmp/pip-req-build-flcbpbf5/external/glslang

      does not contain a CMakeLists.txt file.


    -- Found PythonLibs: /usr/lib/arm-linux-gnueabihf/libpython3.7m.so (found version "3.7.3")
    CMake Error at python/CMakeLists.txt:2 (add_subdirectory):
      The source directory

        /tmp/pip-req-build-flcbpbf5/python/pybind11

      does not contain a CMakeLists.txt file.


    CMake Error at python/CMakeLists.txt:3 (pybind11_add_module):
      Unknown CMake command "pybind11_add_module".


    -- Configuring incomplete, errors occurred!
```

This did not occur when installing from master branch on git or `v0.6.0` in pypi.  I can investigate a little further another evening. I decided just to skirt the issue by installing from git.

Reply:
No worries - I think for now best to stick with master, as there are quite a few bugs fixed, probably not worth investigating as much as it seems the latest one is the one that works